internal/collect/clusters.go

```diff
@@ -33,29 +38,39 @@ func Clusters(config *load.Config) {
 
 		var clusters []mo.ClusterComputeResource
 		// Reference: https://code.vmware.com/apis/704/vsphere/vim.ClusterComputeResource.html
-		err = cv.Retrieve(
-			ctx,
-			[]string{"ClusterComputeResource"},
-			[]string{"summary", "host", "datastore", "name", "network", "configuration"},
-			&clusters)
+		err = cv.Retrieve(ctx, []string{CLUSTER}, propertiesToRetrieve, &clusters)
 		if err != nil {
-			config.Logrus.WithError(err).Error("failed to retrieve ClusterComputeResource")
+			logger.WithError(err).Error("failed to retrieve ClusterComputeResource")
 			continue
 		}
-		if err := collectTags(config, clusters, config.Datacenters[i]); err != nil {
-			config.Logrus.WithError(err).Errorf("failed to retrieve tags:%v", err)
+
+		if collectTags {
+			_, err = config.TagCollector.FetchTagsForObjects(clusters)
+			if err != nil {
+				logger.WithError(err).Warn("failed to retrieve tags for clusters", err)
+			} else {
+				logger.WithField("seconds", config.Uptime()).Debug("clusters tags collected")
+			}
 		}
-		var refSlice []types.ManagedObjectReference
 
-		for j := 0; j < len(clusters); j++ {
-			config.Datacenters[i].Clusters[clusters[j].Self] = &clusters[j]
-			refSlice = append(refSlice, clusters[j].Self)
+		var clusterRefs []types.ManagedObjectReference
+		for _, cluster := range clusters {
+			if filterByTag && !config.TagCollector.MatchObjectTags(cluster.Reference()) {
+				logger.WithField("cluster", cluster.Name).
+					Debug("ignoring cluster since no tags matched the configured filters")
+				continue
+			}
+
+			config.Datacenters[i].Clusters[cluster.Self] = &cluster
+			clusterRefs = append(clusterRefs, cluster.Self)
 		}
 
-		if config.Args.EnableVspherePerfMetrics && dc.PerfCollector != nil {
-			collectedData := dc.PerfCollector.Collect(refSlice, dc.PerfCollector.MetricDefinition.ClusterComputeResource, performance.FiveMinutesInterval)
+		if config.PerfMetricsCollectionEnabled() {
```

Comment:
This was checking has well if `dc.PerfCollector != nil`. I am not sure it is still needed

Reply:
Actually now there is a fatal error in case the perf collector is not correctly initiated. So you are right, it was not useful anymore!